core/src/main/java/org/apache/calcite/adapter/jdbc/JdbcCatalogSchema.java

```diff
@@ -103,34 +143,25 @@ public static JdbcCatalogSchema create(
     return new JdbcCatalogSchema(dataSource, dialect, convention, catalog);
   }
 
-  private SubSchemaMap computeSubSchemaMap() {
-    final ImmutableMap.Builder<String, Schema> builder =
-        ImmutableMap.builder();
-    @Nullable String defaultSchemaName;
-    try (Connection connection = dataSource.getConnection();
-         ResultSet resultSet =
-             connection.getMetaData().getSchemas(catalog, null)) {
-      defaultSchemaName = connection.getSchema();
-      while (resultSet.next()) {
-        final String schemaName =
-            requireNonNull(resultSet.getString(1),
-                "got null schemaName from the database");
-        builder.put(schemaName,
-            new JdbcSchema(dataSource, dialect, convention, catalog, schemaName));
-      }
+  @Override public Lookup<Table> tables() {
+    return Lookup.empty();
+  }
+
+  @Override public Lookup<? extends Schema> subSchemas() {
```

Comment:
what happens if the subschemas change in the underlying connection?
is there an invalidation protocol?

Reply:
The invalidation is happening after one minute as I described in my comment to class `LoadingCacheLookup`